Alerting docs: adds roles docs (#86261)
grafana/grafana: docs/sources/alerting/set-up/configure-roles/index.md

---
canonical: https://grafana.com/docs/grafana/latest/alerting/set-up/configure-roles/
description: Configure roles and permissions for Grafana Alerting
keywords:
  - grafana
  - alerting
  - set up
  - configure
  - roles and permissions
labels:
  products:
    - oss
title: Configure roles and permissions
weight: 150
refs:
  org-roles:

>     - pattern: /docs/grafana/
>       destination: https://grafana.com/docs/grafana/<GRAFANA_VERSION>/administration/roles-and-permissions#organization-roles

Reviewer: Either we need a pattern for Grafana Cloud:

```suggestion
    - pattern: /docs/grafana/
      destination: https://grafana.com/docs/grafana/<GRAFANA_VERSION>/administration/roles-and-permissions#organization-roles
   - pattern: /docs/grafana-cloud/
      destination: https://grafana.com/docs/grafana/<GRAFANA_VERSION>/administration/roles-and-permissions#organization-roles
```

>     - pattern: /docs/grafana/
>       destination: https://grafana.com/docs/grafana/<GRAFANA_VERSION>/administration/roles-and-permissions#organization-roles

Reviewer: Or since they are the same destination, we can use a more general pattern prefix to catch both cases:

```suggestion
    - pattern: /docs/
      destination: https://grafana.com/docs/grafana/<GRAFANA_VERSION>/administration/roles-and-permissions#organization-roles
```

---

# Configure roles and permissions

A user is any individual who can log in to Grafana. Each user is associated with a role that includes permissions. Permissions determine the tasks a user can perform in the system. For example, the Admin role includes permissions for an administrator to create and delete users.

> For more information, refer to [Organization roles][ref:org-roles]

Reviewer: ```suggestion
For more information, refer to [Organization roles](ref:org-roles).
```

## Manage access using roles

For Grafana OSS, there are three roles: Admin, Editor, and Viewer.

Details of the roles and the access they provide for Grafana Alerting are below.

| Role   | Access                                                                                                                                                                    |
| ------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| Admin  | Write access to alert rules, notification resources (notification API, contact points, templates, time intervals, notification policies, and silences), and provisioning. |
| Editor | Write access to alert rules, notification resources (notification API, contact points, templates, time intervals, notification policies, and silences), and provisioning. |
| Viewer | Read access to alert rules, notification resources (notification API, contact points, templates, time intervals, notification policies, and silences).                    |

## Assign roles

To assign roles, admins need to complete the following steps.

> 1. Navigate to **Administration** -> **Users and access** -> **Users, Teams, or Service Accounts**.

Reviewer: Yesterday I found out that we have style guidance to just use the `>` symbol: https://grafana.com/docs/writers-toolkit/write/style-guide/style-conventions/#bold:~:text=Use%20bold%20to%20indicate%20paths%20within%20a%20web%20application%2C%20and%20greater%2Dthan%20symbols%20(%3E)%20to%20indicate%20path%20separators.%20For%20example%3A

```suggestion
1. Navigate to **Administration** > **Users and access** > **Users, Teams, or Service Accounts**.
```

1. Search for the user, team or service account you want to add a role for.
1. Add the role you want to assign.

## Manage access using folder permissions

You can further customize access for alert rules, simplified alert routing, and provisioning by assigning permissions to individual folders.

> This prevents every user from having access to modify all alert rules and gives them access to the folders with the alert rules they are working on.

Reviewer: nit: tone (https://grafana.com/docs/writers-toolkit/write/style-guide/voice-tone-guidelines/#tone)
```suggestion
This prevents every user from having access to modify all alert rules and gives them access to the folders with the alert rules they're working on.
```

> For example, if you are using simplified alert routing and adding contact points to your alert rules, it also helps you avoid the scenario where someone from another team accidentally removes the wrong notification policy or adds a competing one, and all of a sudden you stop getting your notifications.

Reviewer 2: This example seems complicated, and the previous paragraph explains this.  I suggest removing this example.

> In this case, you would assign the **Viewer** role and then add **Editor** permission to the folder. Adding the **Editor** permission to the folder does not overwrite the **Viewer** role.

Reviewer: nit: tone (https://grafana.com/docs/writers-toolkit/write/style-guide/voice-tone-guidelines/#tone)

```suggestion
In this case, you would assign the **Viewer** role and then add **Editor** permission to the folder.
Adding the **Editor** permission to the folder doesn't overwrite the **Viewer** role.
```

Details on the adding folder permissions as well as roles and the access that provides for Grafana Alerting is below.

| Role   | Folder permission | Access                                                                                    |
| ------ | ----------------- | ----------------------------------------------------------------------------------------- |
| Admin  | -                 | Write access to alert rules in all folders.                                               |
| Editor | -                 | Write access to alert rules in all folders.                                               |

> | Viewer | Admin             | Write access to alert rules **only** in the folders where the Admin permission is added.  |

Reviewer 2: ```suggestion
| Viewer | Admin             | Read access to alert rules in all folders. Write access to alert rules **only** in the folders where the Admin permission is added.  |
```

> | Viewer | Editor            | Write access to alert rules **only** in the folders where the Editor permission is added. |

Reviewer 2: ```suggestion
| Viewer | Editor            | Read access to alert rules in all folders. Write access to alert rules **only** in the folders where the Editor permission is added. |
```

> | Viewer | Viewer            | Read access to alert rules.                                                               |

Reviewer 2: ```suggestion
| Viewer | Viewer            | Read access to alert rules in all folders.                                                               |
```

> {{% admonition type="note" %}}
> You cannot use folders to customize access to notification resources.
> {{% /admonition %}}

Reviewer: Same behavior but `{{<` is consistent with Writers' Toolkit (https://grafana.com/docs/writers-toolkit/write/shortcodes/#admonition)

nit: tone (https://grafana.com/docs/writers-toolkit/write/style-guide/voice-tone-guidelines/#tone)

```suggestion
{{< admonition type="note" >}}
You can't use folders to customize access to notification resources.
{{< /admonition >}}
```
